### ISSUE TITLE
Port Network Solver Tool

### ISSUE THREAD

**Issue by geographika** [meggsimum/cpsi-mapview#74]
Port the OL2 based tool at https://github.com/compassinformatics/mapview/blob/master/src/gis/Solver.js

This depends on: 
- https://github.com/compassinformatics/mapview/blob/master/src/gis/ServiceToolMixin.js

This tool relies on a backend service. Test JSON responses will be specified and example test files added to this ticket. A demo of the tool in action can be provided.

**Comment by geographika** [comment]
@weskamm - excellent, this works really well - thanks. 
One piece of functionality that is missing is the ability to click on a point in the path to remove a node. This allows users to reroute with the tool without having to start from scratch again. 
Example screencast from the old system: https://drive.google.com/file/d/12gVY-cUvKFZ9J568Vr1nQWsq7__qfqpw/view

Is this something that can be added to the tool?

**Comment by geographika** [comment]
Quick question on using more specialised versions of these tools - I presume the standard approach of sub-classing the DigitizeButton controller and view is the recommended approach, and override onResponseFeatures?

**Comment by geographika** [comment]
A couple of final comments:

1. Can the point be removed by default if it does not find a road?

![image](https://user-images.githubusercontent.com/490840/56738736-7f848c00-676d-11e9-92d1-745f03a63502.png)

2. If the Ajax request is cancelled or timesout (rather than simply the request failing), then the "Loading" gif is never removed. Is this something that can be handled in Ext.Ajax.request ?

I could recreate this by restarting the web server and creating a large polygon as in the screencast below:

https://drive.google.com/file/d/13o3dLAYDrBZsWjVtPJ-IETwLtyPs_QJ1/view




https://drive.google.com/file/d/13o3dLAYDrBZsWjVtPJ-IETwLtyPs_QJ1/view

**Comment by annarieger** [comment]
Both issues (not found road points and eternal loading mask) are fixed in #86. Please take a look @geographika

**Comment by geographika** [comment]
@annarieger - thanks for these fixes - confirmed as fixed locally.
